koocook_core/models/review.py

```diff
@@ -60,6 +60,60 @@ def item_reviewed(self, obj):
                              type(self.reviewed_recipe or self.reviewed_post or self.reviewed_comment),
                              type(obj))) from e.__context__
 
+class Comment(models.Model):
+    author = models.ForeignKey(
+        'koocook_core.Author',
+        on_delete=models.PROTECT,
+    )
+    date_published = models.DateTimeField()
+    body = models.TextField()
+    aggregate_rating = models.OneToOneField(
+        'koocook_core.AggregateRating',
+        on_delete=models.PROTECT,
+    )
+    # item_reviewed = models.URLField()
+    reviewed_recipe = models.ForeignKey(
+        'koocook_core.Recipe',
+        on_delete=models.CASCADE,
+        null=True,
+        blank=True,
+    )
+    reviewed_post = models.ForeignKey(
+        'koocook_core.Post',
+        on_delete=models.CASCADE,
+        null=True,
+        blank=True,
+    )
+    reviewed_comment = models.ForeignKey(
+        'self',
+        on_delete=models.CASCADE,
+        null=True,
+        blank=True,
+    )
+
+    @property
+    def item_reviewed(self):
+        return self.reviewed_recipe or self.reviewed_post or self.reviewed_comment
+
+    @item_reviewed.setter
+    def item_reviewed(self, obj):
+        from koocook_core.models.recipe import Recipe
+        from koocook_core.models.post import Post
+        try:
+            if self.reviewed_recipe:
+                assert isinstance(obj, Recipe)
+                self.reviewed_recipe = obj
+            elif self.reviewed_post:
+                assert isinstance(obj, Post)
+                self.reviewed_post = obj
+            elif self.reviewed_comment:
+                assert isinstance(obj, self.__class__)
+                self.reviewed_comment = obj
+        except AssertionError as e:
+            raise TypeError('item_reviewed must be of the correct type '
+                            '\'{}\' not \'\''.format(
+                             type(self.reviewed_recipe or self.reviewed_post or self.reviewed_comment),
+                             type(obj))) from e.__context__
```

Comment:
Why re-declare `Comment`?

Reply:
Duplicated code as a result of an editor's resolution